source/tests/timezone_cape_verde_test.cpp

```diff
@@ -72,6 +72,22 @@ int main(int argc, const char* const argv[]) {
         ("F", "01:30"_t)
         ("G", "02:00"_t);
 
+    /* The X line is used to test first/last departure of a line
+     * */
+    b.vj("X", "11100111").uri("X:vj1")("X_S1", "08:00"_t)("X_S2", "16:00"_t)("X_S3", "24:30"_t).make();
+    b.vj("X", "11100111").uri("X:vj1")("X_S1", "09:00"_t)("X_S2", "17:00"_t)("X_S3", "25:30"_t).make();
+    b.vj("X", "11100111").uri("X:vj1")("X_S1", "10:00"_t)("X_S2", "18:00"_t)("X_S3", "26:30"_t).make();
+
+    b.vj("X", "00011000").uri("X:vj2")("X_S1", "08:05"_t)("X_S2", "16:00"_t)("X_S3", "24:45"_t).make();
+    b.vj("X", "00011000").uri("X:vj2")("X_S1", "09:05"_t)("X_S2", "17:00"_t)("X_S3", "25:45"_t).make();
+    b.vj("X", "00011000").uri("X:vj2")("X_S1", "10:05"_t)("X_S2", "18:00"_t)("X_S3", "26:45"_t).make();
+    auto line_X = b.lines.find("X")->second;
+
+    // opening time should be the earliest departure of all vjs of the line
+    line_X->opening_time = boost::make_optional(boost::posix_time::duration_from_string("08:00"));
```

Comment:
This may deserve tests on another TZ to check it works fine (with DST? :-/ ).

Reply:
we won't handle DST for this feature, it's kind of hard to do(both implementation and test) and the benefit is minor. I'd prefer wait for the bug that will be reported by the client with a concrete example